### PR TITLE
optimization: Replaced strings.Index with strings.Cut

### PR DESCRIPTION
**Pull request by chir4gm**
modified:   cmd/main.go
	modified:   listeners.go
	modified:   modules/caddyhttp/reverseproxy/addresses.go
	modified:   modules/caddyhttp/reverseproxy/reverseproxy.go
	modified:   modules/caddyhttp/rewrite/rewrite.go

to fix Issue #4928

### CODE REVIEW

**Comment by francislavoie** [comment]
Sorry, this has already been worked on here: https://github.com/caddyserver/caddy/pull/4930!

Thanks anyways.

**Comment by mholt** [comment]
This PR appears to modify different files.

File: listeners.go
Comment (mholt) [inline listeners.go:501]: This change has already been made in another PR.

File: modules/caddyhttp/reverseproxy/addresses.go
Comment (mholt) [inline modules/caddyhttp/reverseproxy/addresses.go:83]: This change has already been made in another PR.

File: modules/caddyhttp/rewrite/rewrite.go
Comment (mholt) [inline modules/caddyhttp/rewrite/rewrite.go:201]: Restore the comments here, I think they are valuable

File: modules/caddyhttp/rewrite/rewrite.go
Comment (mholt) [inline modules/caddyhttp/rewrite/rewrite.go:206]: This empty line should probably remain since the comment above doesn't apply to these lines

**Review by mholt — CHANGES_REQUESTED** [review]
Thanks for the contribution! This mostly looks good; just a few nits and a merge conflict to take care of first.

**Comment by chir4gm** [comment]
Hi, I'm new to OSS, but I have 'force' pushed a new commit with the requested changes after using `git reset` and modifying the `HEAD` commit on the `stringsFix` branch. Please tell me what to do in case something is wrong.

**Comment by mholt** [comment]
@chir4gm Welcome to open source!

If you look at this page and scroll down you'll see "Conflicting files" and it has a link that says "command line" -- if you click that, instructions will appear about how to resolve conflicts.

Basically, changes were made to the same code in where you want to merge to, so you need to update your branch to figure out which changes should be the final result: yours or theirs.

**Comment by chir4gm** [comment]
Got it! I think the other conflicting commit had semantically better variables. I hope the conflicts have been correctly resolved.

**Comment by mholt** [comment]
Looks much better!

Now make sure to indent using tabs and not spaces. :)

**Comment by chir4gm** [comment]
Hi, I manually made the formatting change. Quick question: When merging a pull request, can maintainers merge all of my commits from my branch into a single one?

**Review by mholt — APPROVED** [review]
> Hi, I manually made the formatting change. Quick question: When merging a pull request, can maintainers merge all of my commits from my branch into a single one?

Yes! We can "Squash and merge" and that's usually what we prefer to do here at the Caddy project, so you don't necessarily have to do that. There are a few occasions on larger PRs where multiple things are happening that we want to keep individual commits but that's pretty rare.

I'll merge this when CI is done. Thanks!

**Pull request merged**
